src/libstd/net/parser.rs

```diff
@@ -374,18 +374,19 @@ impl FromStr for SocketAddr {
 ///
 /// # Potential causes:
 ///
-/// `AddrParseError` may be thrown because of wrong resource typing.
+/// `AddrParseError` may be thrown because the provided string does not parse as the given type,
+/// often because it includes informations only handled by a different address type.
 ///
 /// ```
 /// use std::net::IpAddr;
 /// let _foo: IpAddr = "127.0.0.1:8080".parse().unwrap(); // panic!
 /// ```
 ///
-/// [`IpAddr`] doesn't support the port. Use [`SocketAddr`] instead.
+/// [`IpAddr`] doesn't handle the port. Use [`SocketAddr`] instead.
 ///
 /// ```
 /// use std::net::SocketAddr;
-/// // That's it. `rustc` isn't angry anymore.
+/// // No problem, `panic!` message has disappear.
```

Comment:
Nit: "has disappeared" rather than "has disappear". Also, need a "the" before the `panic!`.

Looks good otherwise, thanks for the update!